### PR TITLE
Compatible ui new restructure

### PR DESCRIPTION
**Pull request by smebrahimi**
This issue #196

### CODE REVIEW

File: src/components/Email/DraftEmail.vue [hunk 1]
```diff
@@ -41,24 +35,38 @@ export default {
   },
   mounted() {
     if (!this.get_email_date) return this.$router.push({ name: 'EmailsAdd', params: { step: 1 } });
+    this.fetch_available_articles({ jumpStartId: this.get_selected_jumpstart.id });
```
Comment (atharva3010) [inline src/components/Email/DraftEmail.vue:38]: I think it's good to fetch the articles when we select the date, if SelectEmailDate.vue is still in the UI flow

File: src/components/Email/DraftEmail.vue [hunk 2]
```diff
@@ -41,24 +35,38 @@ export default {
...
   computed: {
-    ...mapGetters('email', ['get_email_date', 'get_selected_articles', 'get_articles'])
+    ...mapGetters('email', [
+      'get_email_date',
+      'get_selected_articles',
+      'get_available_articles',
+      'get_selected_jumpstart'
+    ]),
+    get_all_articles() {
+      return [...this.get_selected_articles, ...this.get_available_articles];
```
Comment (atharva3010) [inline src/components/Email/DraftEmail.vue:68]: this may duplicate some items

File: src/store/modules/email.js [hunk 3]
```diff
@@ -50,13 +56,20 @@ export default {
...
+    },
     update_preview_link({ state }, payload) {
       if (payload === '') {
         state.preview_link = payload;
         return false;
       }
       const base = `${BASE_API}/api/v0/volkswagen/jumpstart/${state.selected_jumpstart.id}/preview`;
       const url = new URL(base);
+      state.selected_articles.forEach(article => {
+        url.searchParams.append('articleIds', article.id);
```
Comment (atharva3010) [inline src/store/modules/email.js:71]: The api supports this? In the swagger, I don't see support for articleIds on preview link, is this a new addition?

File: src/components/Email/DraftEmail.vue (hunk 2 above)
Comment (atharva3010) [reply src/components/Email/DraftEmail.vue:68]: @mort3za cleared this doubt

**Comment by mort3za** [comment]
I see a problem in date selector, There is two `11 May 2020`. That's coming from API @ctyar 

image

**Comment by mort3za** [comment]
Merged to prevent bottleneck. Please check the comments and create a new PR if it was necessary. @smebrahimi

File: src/store/modules/email.js (hunk 3 above)
Comment (smebrahimi) [reply src/store/modules/email.js:71]: Screenshot from 2020-05-09 22-00-05
Yes

**Pull request merged**